src/wp-includes/block-template.php

```diff
@@ -5,6 +5,32 @@
  * @package WordPress
  */
 
+/**
+ * Adds necessary filters to use 'wp_template' posts instead of theme template files.
+ *
+  * @access private
+  * @since 5.9.0
+ */
+function _add_template_loader_filters() {
+	if ( ! current_theme_supports( 'block-templates' ) ) {
+		return;
+	}
+
+	foreach ( _get_template_type_slugs() as $template_type ) {
+		if ( 'embed' === $template_type ) { // Skip 'embed' for now because it is not a regular template type.
+			continue;
+		}
+		add_filter( str_replace( '-', '', $template_type ) . '_template', 'locate_block_template', 20, 3 );
+	}
+
+	// Request to resolve a template.
+	if ( isset( $_GET['_wp-find-template'] ) ) {
+		add_filter( 'pre_get_posts', '_resolve_template_for_new_post' );
+	}
+}
+
+add_action( 'wp_loaded', '_add_template_loader_filters' );
```

Comment:
This line of code needs to move to `wp-includes/default-filters.php` as @noisysocks pointed out here https://github.com/WordPress/wordpress-develop/pull/1997#discussion_r761653397.

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/1997/commits/c97a6e640ba085e513da646db8274a7e3690ccd3 and https://github.com/WordPress/wordpress-develop/pull/1997/commits/a19dbb6bbf62137ff3d0794a7492c0edf3a45a03.